src/AwsS3Adapter.php

```diff
@@ -438,7 +448,7 @@ public function listContents($dirname = '', $recursive = false)
      */
     protected function normalizeResponse(array $object, $path = null)
     {
-        $result = ['path' => $path ?: $this->removePathPrefix($object['Key'])];
+        $result = ['path' => $path ?: $this->removePathPrefix(isset($object['Key']) ? $object['Key'] : $object['Prefix'])];
```

Comment:
If content is listed non-recursively prefixes are listed without `Key` attribute. The path of the directory is stored as `Prefix`. Directory level doesn't matter here.